gobblin-core-base/src/main/java/org/apache/gobblin/source/extractor/extract/LongWatermark.java

```diff
@@ -29,20 +29,23 @@
 import org.apache.gobblin.source.extractor.Watermark;
 
 import lombok.EqualsAndHashCode;
-import lombok.Getter;
 import lombok.Setter;
 import lombok.ToString;
 
 @ToString
 @EqualsAndHashCode
-public class LongWatermark implements ComparableWatermark {
+public class LongWatermark implements ComparableWatermark<Long> {
 
   private static final Gson GSON = new Gson();
 
-  @Getter
   @Setter
   private long value;
 
+  @Override
+  public Long getValue(){
```

Comment:
We can probably add the explicit java doc for explicit use of long as class variable but Long as return value.

```
Returns a Long object due to Java generics' requirement for object types.
The underlying variable is maintained as a primitive long to optimize performance for mathematical operations.
```